### PR TITLE
Add aliases to 2021-04-30-preview swagger for Azure search

### PR DESCRIPTION
**Pull request by robertklee**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  
February at the earliest.

  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  
February at the earliest.

  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceDeleteAlias.json
Comment (BevLoh) [inline specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceDeleteAlias.json:9]: Does Index also return 404? This doesn't align with idempotent principles but if you're following existing patterns, unfortunately we'll have to accept it.

File: specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json
Comment (BevLoh) [inline specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json:1992]: Is there missing update for alias count in the service stats?

File: specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceCreateAlias.json
Comment (BevLoh) [inline specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceCreateAlias.json:2]: Is there missing update for alias count in the service stats example?

File: specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceDeleteAlias.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceDeleteAlias.json:9]: Unfortunately yes, this is following Index's existing pattern:
```json
      "delete": {
        "tags": [
          "Indexes"
        ],
        "operationId": "Indexes_Delete",
        "x-ms-examples": {
          "SearchServiceDeleteIndex": {
            "$ref": "./examples/SearchServiceDeleteIndex.json"
          }
        },
        "description": "Deletes a search index and all the documents it contains. This operation is permanent, with no recovery option. Make sure you have a master copy of your index definition, data ingestion code, and a backup of the primary data source in case you need to re-build the index.",
        "externalDocs": {
          "url": "https://docs.microsoft.com/rest/api/searchservice/Delete-Index"
        },
        "parameters": [
          {
            "name": "indexName",
            "in": "path",
            "required": true,
            "type": "string",
            "description": "The name of the index to delete."
          },
          {
            "$ref": "#/parameters/ClientRequestIdParameter"
          },
          {
            "$ref": "#/parameters/IfMatchParameter"
          },
          {
            "$ref": "#/parameters/IfNoneMatchParameter"
          },
          {
            "$ref": "#/parameters/ApiVersionParameter"
          }
        ],
        "x-ms-request-id": "request-id",
        "responses": {
          "204": {
            "description": ""
          },
          "404": {
            "description": ""
          },
          "default": {
            "description": "Error response.",
            "schema": {
              "$ref": "#/definitions/SearchError"
            }
          }
        }
      },
```

File: specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceCreateAlias.json
Comment (robertklee) [reply specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/examples/SearchServiceCreateAlias.json:2]: Good catch, thanks!

**Comment by robertklee** [comment]
Breaking change failure should be ok since this is a preview version. tagging @dereklegenzoff for visibility.

**Comment by robertklee** [comment]
Toggling to full PR, please do not merge for now. I'd like to get the PR to the state where it's ready to merge, and merge once the necessary feature code is deployed.

**Review by tjprescott — APPROVED** [review]

**Comment by robertklee** [comment]
@tjprescott, it's ready for merge. Thanks for your help throughout this PR!

File: specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json
Comment (Mohit-Chakraborty) [inline specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json:7825]: Is the comment outdated? Why does it take in an array?

Comment (tjprescott) [reply specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json:7825]: This is a common pattern for services who want to future-proof their APIs. An array that only accepts a single value today may accept multiple values tomorrow and that would not be a breaking change, but changing from a single type to an array would be.

Comment (dereklegenzoff) [reply specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json:7825]: Travis is spot on. Eventually we may want to support customers being able to specify multiple indexes for a single alias.

Comment (Mohit-Chakraborty) [reply specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json:7825]: Another use case - multiple aliases for a single index. Is that a less common scenario?

Comment (robertklee) [reply specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json:7825]: Thanks for bringing those up. My take on the your questions:
> Just wondering if the comment should be clearer to say something like "Only one index name may be specified currently".

The *currently* would imply we have plans to add it in the future. Since we aren't sure of that, I think it's best to omit it.

> Also, I hope we update the comment when in a later update the service behavior changes.

Yes, we will update it in the future if/when the behavior changes.

> Another use case - multiple aliases for a single index. Is that a less common scenario?

This is allowed. The customer would just need to create multiple aliases, each mapping to the same index.

**Pull request merged**
